pytest/tests/mirror/test.py

```diff
@@ -0,0 +1,316 @@
+                        rpc_node_names=[],
+                        chain_id='shardnet',
+                        append=True,
+                        epoch_length=20,
+                        node_pks=None,
+                        increasing_stakes=0.0,
+                        num_seats=len(SHARDNET_VALIDATORS))
+    return init_shardnet_dirs(neard)
+
+
+def init_mirror_dir(home, source_boot_node):
+    mkdir_clean(dot_near() / 'mirror')
+    os.rename(home, dot_near() / 'mirror/source')
+    ordinal = NUM_VALIDATORS
+    with open(dot_near() / 'mirror/source/config.json', 'r') as f:
+        config = json.load(f)
+        config['network']['boot_nodes'] = source_boot_node.addr_with_pk()
+        config['network']['addr'] = f'0.0.0.0:{24567 + 10 + ordinal}'
+        config['rpc']['addr'] = f'0.0.0.0:{3030 + 10 + ordinal}'
+    with open(dot_near() / 'mirror/source/config.json', 'w') as f:
+        json.dump(config, f)
+
+
+def mirror_cleanup(process):
+    process.send_signal(signal.SIGINT)
+    try:
+        process.wait(5)
+    except:
+        logger.error('can\'t kill mirror process')
+
+
+def start_mirror(near_root, target_home, boot_node):
+    env = os.environ.copy()
+    env["RUST_LOG"] = "actix_web=warn,mio=warn,tokio_util=warn,actix_server=warn,actix_http=warn," + env.get(
+        "RUST_LOG", "debug")
+    with open(dot_near() / 'mirror/stdout', 'ab') as stdout, \
+        open(dot_near() / 'mirror/stderr', 'ab') as stderr:
+        process = subprocess.Popen([
+            os.path.join(near_root, 'mirror'), 'run', "--source-home",
+            dot_near() / 'mirror/source/', "--target-home", target_home,
+            '--secret-file', target_home / 'mirror-secret.json'
+        ],
+                                   stdin=subprocess.DEVNULL,
+                                   stdout=stdout,
+                                   stderr=stderr,
+                                   env=env)
+    logger.info("Started mirror process")
+    atexit.register(mirror_cleanup, process)
+    with open(target_home / 'config.json', 'r') as f:
+        config = json.load(f)
+        config['network']['boot_nodes'] = boot_node.addr_with_pk()
+    with open(target_home / 'config.json', 'w') as f:
+        json.dump(config, f)
+    return process
+
+
+# we'll test out adding an access key and then sending txs signed with it
+# since that hits some codepaths we want to test
+def send_add_access_key(node, creator_key, nonce, block_hash):
+    k = key.Key.from_random('test0')
+    action = transaction.create_full_access_key_action(k.decoded_pk())
+    tx = transaction.sign_and_serialize_transaction(
+        'test0', nonce, [action], base58.b58decode(block_hash.encode('utf8')),
+        'test0', creator_key.decoded_pk(), creator_key.decoded_sk())
+    node.send_tx(tx)
+    return k
+
+
+def main():
+    config_changes = {}
+    for i in range(NUM_VALIDATORS + 1):
+        config_changes[i] = {"tracked_shards": [0, 1, 2, 3], "archive": True}
```

Comment:
It probably would be more consistent to set this to `[0]`, like all configs on all nodes currently do.
But I like that the list of shards is defined explicitly. Let's keep it like that.